### PR TITLE
[ASR PASS]: Support printing multiple lists at a time

### PR DESCRIPTION
**Pull request by Smit-create**
Part of #1410

### CODE REVIEW

File: src/libasr/pass/print_list.cpp [hunk 1]
```diff
@@ -52,148 +52,186 @@ class PrintListVisitor
...
+                al, loc, s2c(al, "]"), str_type_len_1));
+
+        std::string list_iter_var_name;
+        ASR::symbol_t *list_iter_variable;
+        ASR::expr_t *list_iter_var;
+        {
+            list_iter_var_name =
+                current_scope->get_unique_name("__list_iterator");
+            list_iter_variable =
+                ASR::down_cast<ASR::symbol_t>(ASR::make_Variable_t(
```
Comment (czgdp1807) [inline src/libasr/pass/print_list.cpp:98]: Use the following function, 

https://github.com/lcompilers/lpython/blob/ed807bbf11ec00e42a4a144adef48e6c67589ffe/src/libasr/pass/pass_utils.h#L53-L54

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: integration_tests/print_02.py [hunk 2]
```diff
@@ -11,10 +11,13 @@ def f():
     print(b)
     print(c)
     print(d)
+    print(a, a, b, c, d)
```
Comment (czgdp1807) [inline integration_tests/print_02.py:14]: What about `print(a, a, 1, b, c, 1.1, d)`?

Comment (Smit-create) [reply integration_tests/print_02.py:14]: Yep, does work:
```py
from ltypes import i32, f64

def f():
    a: list[str] = ["ab", "abc", "abcd"]
    b: list[i32] = [1, 2, 3, 4]
    c: list[f64] = [1.23 , 324.3 , 56.431, 90.5, 34.1]
    d: list[i32] = []

    print(a, a, 1, b, c, 1.1, d)

f()
```
Output:
```console
% lpython a.py 
['ab', 'abc', 'abcd']
['ab', 'abc', 'abcd']
1
[1, 2, 3, 4]
[1.22999999999999998e+00, 3.24300000000000011e+02, 5.64309999999999974e+01, 9.05000000000000000e+01, 3.41000000000000014e+01]
1.10000000000000009e+00
[]
```

Comment (czgdp1807) [reply integration_tests/print_02.py:14]: Great. Add it as a test.

**Review by czgdp1807 — APPROVED** [review]

**Pull request merged**